### PR TITLE
Mark deprecated kubelet-timeout

### LINKED ISSUE #102052: ApiServer `--kubelet-timeout` does nothing
#### What happened:
Configured `--kubelet-timeout` but no effect.

#### Anything else we need to know?:
Check the code and found the `--kubelet-timeout` does nothing.

https://github.com/kubernetes/kubernetes/blob/1d38084930b56be9b5830c20bd5b464b1f955047/cmd/kube-apiserver/app/options/options.go#L252-L253
https://github.com/kubernetes/kubernetes/blob/1d38084930b56be9b5830c20bd5b464b1f955047/pkg/kubelet/client/kubelet_client.go#L87-L123

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?
/kind bug
/kind documentation

#### Which issue(s) this PR fixes:

Fixes #102052

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: cmd/kube-apiserver/app/options/options.go [hunk 1]
```diff
@@ -250,7 +250,8 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
 	fs.MarkDeprecated("kubelet-read-only-port", "kubelet-read-only-port is deprecated and will be removed.")
 
 	fs.DurationVar(&s.KubeletConfig.HTTPTimeout, "kubelet-timeout", s.KubeletConfig.HTTPTimeout,
-		"Timeout for kubelet operations.")
+		"DEPRECATED: Timeout for kubelet operations.")
```
Comment (swatisehgal) [inline cmd/kube-apiserver/app/options/options.go:253]: I think we should specify the release here as well.

File: cmd/kube-apiserver/app/options/options.go [hunk 2]
```diff
@@ -250,7 +250,8 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
...
+	fs.MarkDeprecated("kubelet-timeout", "kubelet-timeout is deprecated and will be removed.")
```
Comment (swatisehgal) [inline cmd/kube-apiserver/app/options/options.go:254]: I think we should specify the release here as well.

**Comment by swatisehgal** [comment]
/hold

This will need further investigation to ensure that this is no longer needed and we want to go ahead with the deprecation of this flag.

**Comment by n4j** [comment]
@swatisehgal Hey Swathi, did you get a chance to investigate it further?

**Comment by swatisehgal** [comment]
@n4j Looking into it. Thanks for the reminder.

**Comment by swatisehgal** [comment]
@wzshiming Thanks for flagging this issue and bringing this to our notice. Is there a reason we are deprecating the flag rather than aiming for a fix? I would be inclined that we find a way fix the functionality of the flag rather than deprecating it. Having said that if there is a strong reason that it is not possible to fix it, we can deprecate it.

**Pull request closed without merging**